File: languages/r/prime.R
is_prime <- function(n) {
  if (n <= 1) {
    return(FALSE)
  }
  if (n == 2) {
    return(TRUE)
  }
  if (n %% 2 == 0) {
    return(FALSE)
  }
  
  for (i in seq(3, floor(sqrt(n)), by=2)) {
    if (n %% i == 0) {
      return(FALSE)
    }
  }
  return(TRUE)
}

cat("Enter a number to check if it's prime: ")
number <- as.integer(readLines("stdin", n=1))

if (is_prime(number)) {
  cat(number, "is a prime number.\n")
} else {
  cat(number, "is not a prime number.\n")
}

File: languages/haskell/prime.hs
import System.IO

main :: IO ()
main = do
    putStrLn "Enter a number to check if it's prime: "
    input <- getLine
    let n = read input :: Integer
    if isPrime n then
        putStrLn (show n ++ " is a prime number.")
    else
        putStrLn (show n ++ " is not a prime number.")

isPrime :: Integer -> Bool
isPrime n
    | n <= 1    = False
    | n <= 3    = True
    | n `mod` 2 == 0 || n `mod` 3 == 0 = False
    | otherwise = null [ x | x <- [5,7..(floor . sqrt . fromIntegral) n], n `mod` x == 0 ]
